pandas/tests/groupby/test_groupby.py

```diff
@@ -2419,3 +2419,27 @@ def test_rolling_wrong_param_min_period():
     result_error_msg = r"__init__\(\) got an unexpected keyword argument 'min_period'"
     with pytest.raises(TypeError, match=result_error_msg):
         test_df.groupby("name")["val"].rolling(window=2, min_period=1).sum()
+
+
+@pytest.mark.parametrize(
+    "func, expected, dtype, result_dtype",
+    [
+        ("sum", [5, 7, 9], int, "int64"),
+        ("std", [4.5 ** 0.5] * 3, int, float),
+        ("var", [4.5] * 3, int, float),
+        ("sum", [5, 7, 9], "Int64", "Int64"),
+        # result_dtype should ideally be Float64
+        # GH#43330
+        ("std", [4.5 ** 0.5] * 3, "Int64", float),
+        ("var", [4.5] * 3, "Int64", "Float64"),
```

Comment:
this actually casts to Float64? what were the results in 1.2.5? we should match them.

Reply:
Yeah. 1.2.5 was `int64, float64, int64`
I will try to do that (didn't get a chance yet)

Reply:
great, yeah we want to have an exact match to 1.2.5; this can be changed going forwawrd for 1.4, but not on a regression patch.